src/hooks/screen/useChatScreenHook.ts

```diff
@@ -317,6 +310,19 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
     }
   }, [localDb, otherListener[`chat_${selectedChannel?.id}`], selectedChannel]);
 
+  React.useEffect(() => {
+    return () => {
+      setChats([]);
+    };
+  }, []);
+
+  React.useEffect(() => {
+    if (getAllMessages.data) {
+      setChats(getAllMessages.data);
+      setIsLoadingFetchAllMessage(false);
+    }
+  }, [getAllMessages.data]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The useEffect hook at lines 313-317 is used to clear the chats state when the component unmounts. This is good practice as it prevents potential memory leaks. The second useEffect hook at lines 319-324 updates the chats state and isLoadingFetchAllMessage state whenever getAllMessages.data changes. This is also good practice as it ensures that the state is always up-to-date with the latest data.

However, it's important to note that if getAllMessages.data is large, this could potentially cause performance issues as the entire array is being set in state every time it changes. Consider using pagination or infinite scrolling to load and display the data in chunks instead of all at once.
```diff
  React.useEffect(() => {
    if (getAllMessages.data) {
      // Use pagination or infinite scrolling here
      setChats(getAllMessages.data);
      setIsLoadingFetchAllMessage(false);
    }
  }, [getAllMessages.data]);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->